java/src/jmri/jmrit/display/layoutEditor/LayoutTurnout.java

```diff
@@ -115,63 +115,149 @@
  * required to be able to correctly interpret the use of signal heads.
  *
  * @author Dave Duchamp Copyright (c) 2004-2007
- * @author George Warner Copyright (c) 2017-2019
+ * @author George Warner Copyright (c) 2017-2020
  */
 public class LayoutTurnout extends LayoutTrack {
 
-    // defined constants - turnout types
-    //public static final int NONE = 0;
-    //public static final int RH_TURNOUT = 1;
-    //public static final int LH_TURNOUT = 2;
-    //public static final int WYE_TURNOUT = 3;
-    //public static final int DOUBLE_XOVER = 4;
-    //public static final int RH_XOVER = 5;
-    //public static final int LH_XOVER = 6;
-    //public static final int SINGLE_SLIP = 7;    // used for LayoutSlip which extends this class
-    //public static final int DOUBLE_SLIP = 8;    // used for LayoutSlip which extends this class
-    public enum TurnoutType {
-        NONE,
-        RH_TURNOUT,
-        LH_TURNOUT,
-        WYE_TURNOUT,
-        DOUBLE_XOVER,
-        RH_XOVER,
-        LH_XOVER,
-        SINGLE_SLIP, // used for LayoutSlip which extends this class
-        DOUBLE_SLIP     // used for LayoutSlip which extends this class
-    }
-
     /**
-     * Returns true if this is a turnout (not a crossover or slip)
-     *
-     * @param type the turnout type
-     * @return boolean true if this is a turnout
+     * TurnoutType enum
+     * <p>
+     * Note: constructor is NAME(xmlValue). The previous xml save code used the
+     * hard-coded xml int values. With introduction of this enum that code has
+     * been changed to save those values as the TurnoutType enum name string
+     * instead. For backwards compatibility the xml int values are still
+     * recognized on load. They should NOT be used going forward so that some
+     * point they can be removed.
+     * <p>
+     * Note also that there are placeholders for values that were previously
+     * used as offsets from base values. These now have to exist as enums so
+     * that the xml loading code can resolve them.
      */
-    public static boolean isTurnoutTypeTurnout(TurnoutType type) {
-        return (type == TurnoutType.RH_TURNOUT
-                || type == TurnoutType.LH_TURNOUT
-                || type == TurnoutType.WYE_TURNOUT);
-    }
+    public enum TurnoutType {
+        NONE(0),
```

Comment:
Please turn this back to a straight enum. That's all that's needed now, and it's on the direct line to what's needed later.

Reply:
The map code is fragile to the ordinal. When we can remove the xml int code we can remove all the ENUM(int) code at that time and switch over to the map. I see no value added and considerable risk switching now.

Reply:
EnumIO use restored